src/libopenrave/kinbodygrab.cpp

```diff
@@ -260,6 +277,35 @@ void Grabbed::ComputeListNonCollidingLinks()
     // }
 }
 
+void Grabbed::_PushNonCollidingLinkPairsForGrabbedBodies(KinBodyPtr& pGrabber,
+                                                         CollisionCheckerBasePtr& pchecker,
+                                                         const KinBody& grabbedBody,
+                                                         const KinBody& otherGrabbedBody)
+{
+    // Find item by (grabbedBody, otherGrabbedBody) pair.
+    const uint64_t indexForTest0 = KinBody::_ComputeEnvironmentBodyIndicesPair(grabbedBody, otherGrabbedBody);
```

Comment:
Why don't we create a rule to how key of `_mapListNonCollidingInterGrabbedLinkPairsWhenGrabbed` is serialized? Rule could be the upper 32 bit's int is smaller than lower 32 bit's (key is `intA | intB << 32` where `intA` is always smaller than `intB`). Then we don't need to search `_mapListNonCollidingInterGrabbedLinkPairsWhenGrabbed` twice.

Reply:
thanks, nice idea!

I fixed the PR.